packaging_automation/tests/test_package_cloud_statistics_collector.py

```diff
@@ -0,0 +1,37 @@
+import os
+from sqlalchemy import text, create_engine
+from ..dbconfig import (Base, db_session, DbParams, db_connection_string)
+from ..package_cloud_statistics_collector import (fetch_and_save_package_cloud_stats, PackageCloudRepos,
+                                                  PackageCloudOrganizations, package_count, PackageCloudDownloadStats)
+import math
+
+DB_USER_NAME = os.getenv("DB_USER_NAME")
+DB_PASSWORD = os.getenv("DB_PASSWORD")
+DB_HOST_AND_PORT = os.getenv("DB_HOST_AND_PORT")
+DB_NAME = os.getenv("DB_NAME")
+PACKAGE_CLOUD_API_TOKEN = os.getenv("PACKAGE_CLOUD_API_TOKEN")
+REPO = PackageCloudRepos.azure
+ORGANIZATION = PackageCloudOrganizations.citusdata
+db_parameters = DbParams(user_name=DB_USER_NAME, password=DB_PASSWORD, host_and_port=DB_HOST_AND_PORT, db_name=DB_NAME)
+PARALLEL_COUNT = 5
+
+
+def test_fetch_and_save_package_cloud_stats():
+    db = create_engine(db_connection_string(db_params=db_parameters, is_test=True))
+    db.execute(text(f'DROP TABLE IF EXISTS {PackageCloudDownloadStats.__tablename__}'))
+    pack_count = package_count(ORGANIZATION, REPO, PACKAGE_CLOUD_API_TOKEN)
+    session = db_session(db_params=db_parameters, is_test=True)
+    index = 0
+    iteration_count = math.ceil(pack_count / PARALLEL_COUNT)
+    page_record_count = iteration_count
+    while index < iteration_count:
```

Comment:
I'm again super confused here.
I think instead of the `iteration_count` we need to put `PARALLEL_COUNT`. Because the `parallel_exec_index` needs to go up to `parallel_count`.
And also `page_record_count` is not something we calculate. We just decide on a number, like 10.

Do you agree with me? Did I get this all wrong?

Reply:
I'm a little bit confused as well. I clarified the logic and hardcoded the page cound to make it more explicit.
``` 
page_record_count = 5
iteration_count = math.ceil(pack_count / page_record_count)
```

Reply:
I'm not convinced 😄. I still think `iteration_count` should be `PARALLEL_COUNT`.
If I understand correctly, right now, the code divides package count by package count per page in `iteration_count = math.ceil(pack_count / page_record_count)`. This technically gives us the page count. So we call the `fetch_and_save_package_cloud_stats` once for every page. But in that function we already go through multiple pages, we don't have to call it for every page.
And in the `package-cloud-download-schedule.yml` we also call `fetch_and_save_package_cloud_stats` `PARALLEL_COUNT` times, which is 10 there.
Actually this would still work. But I think the correct way would be to call `PARALLEL_COUNT` times.

Do you aggree?

Reply:
Fixed